src/nodes.coffee

```diff
@@ -4537,6 +4537,9 @@ exports.For = class For extends While
 
   addBody: (body) ->
     @body = Block.wrap [body]
+    {expressions} = @body
+    if expressions.length
+      @body.locationData ?= mergeLocationData expressions[0].locationData, expressions[expressions.length - 1].locationData
```

Comment:
Ideally this would live inside `Block.wrap()` so that it never returned a `Block` with no location data. But that will have to wait as there are currently other node class objects getting passed to it that don't have location data to use in deriving the new `Block`'s location data